processor/processor.go

```diff
@@ -432,6 +432,7 @@ func (proc *HandleT) Shutdown() {
 var (
 	enablePipelining          bool
 	pipelineBufferedItems     int
+	subJobCount               int
```

Comment:
Why not splitting the 10K jobs into 10K subjobs of 1 job, instead of 5 subjobs of 2K jobs?

Reply:
To make the most efficient use of memory using asynchronous step processing through channels, each of the three channels should receive a singular item (message) instead of the full batch. Anything in between is going to be a compromise. Do you have any metrics to share with regards to how much the throughput decreases using singular items? Even better, would it be too hard for me to setup a test bench locally?
The structs that are being passed around through the channels appear to be complex indeed, holding stuff other than job messages, so I will need some time to study and try to explain why throughput decreases as you mentioned, however I believe it might be worth giving some effort towards this direction.

Reply:
It appears that transformer HTTP calls are being batched for same src+dest key messages during

1. the `process` step for performing trackingPlan validation 
2. the `transform` step for custom & destination-specific transformations

This can explain why throughput decreases drastically when splitting the batch in very small sub-batches.

Reply:
yup